tests/feed/test_feed.py

```diff
@@ -0,0 +1,156 @@
+
+    course.save()
+    return course
+
+
+@pytest.fixture
+def persist_second_course(persist_second_teacher):
+    course = TeacherCourses(teacher_id=persist_second_teacher,
+                            course_name="riding a tricycle",
+                            description="riding a tricycle course for advanced tricycle riders",
+                            price=200,
+                            years_of_experience=5,
+                            difficulty_level="A",
+                            category="SPORT_AND_LEISURE")
+    course.save()
+    return course
+
+
+@pytest.fixture
+def persist_first_student_course(persist_course, persist_user):
+    student_course = StudentCourses(student_id=persist_user,
+                                    teacher_course_id=persist_course,
+                                    status="Confirmed")
+    student_course.save()
+    return student_course
+
+
+@pytest.fixture
+def persist_second_student_course(persist_second_course, persist_second_user):
+    student_course = StudentCourses(student_id=persist_second_user,
+                                    teacher_course_id=persist_second_course,
+                                    status="Confirmed")
+    student_course.save()
+    return student_course
+
+
+@pytest.fixture
+def persist_post_for_first_course(persist_course, persist_user, persist_first_student_course):
+    post = Post(course_id=persist_course,
+                user_id=persist_user,
+                parent_post_id=None,
+                content="this is a root post by a student for first course")
+    post.save()
+    return post
+
+
+@pytest.fixture
+def persist_reply_for_first_course(persist_course, persist_teacher,
+                                   persist_post_for_first_course, persist_second_student_course):
+    post = Post(course_id=persist_course,
+                user_id=persist_teacher,
+                parent_post_id=persist_post_for_first_course,
+                content="this is a reply post by a teacher for message in first course")
+    post.save()
+    return post
+
+
+@pytest.fixture
+def persist_post_for_second_course_student(persist_second_course, persist_second_user):
+    post = Post(course_id=persist_second_course,
+                user_id=persist_second_user,
+                parent_post_id=None,
+                content="this is a root post by a second student for second course")
+    post.save()
+    return post
+
+
+@pytest.mark.django_db
+class TestPost:
+    def test_is_root_post(self, persist_post_for_first_course, persist_reply_for_first_course):
+        assert persist_post_for_first_course.is_root_post()
+        assert not persist_reply_for_first_course.is_root_post()
+
+    def test_post_is_only_reply_to_root_post(self, persist_reply_for_first_course,
+                                             persist_post_for_second_course_student):
+        persist_post_for_second_course_student.parent_post_id = persist_reply_for_first_course
```

Comment:
If we are making a change to the fixture so early on, maybe we don't need a fixture

Reply:
I think this comment I made explains my choice here:
https://github.com/redhat-beyond/HighTeach/pull/56#pullrequestreview-1393841803

Reply:
Personally I don't like that approach that much, because I expect a certain result of the test given the basic fixture, but then it is changed. I would rather, if needed, create another fixture that uses the current one, then using that new fixture instead. it can also have a readable name like "persist_reply_for_first_course_with..."

Reply:
so It's bringing us back to the point of - maybe we don't need the fixtures. OR- maybe we need a `fixture factory`, to produce a fixture object on every test.

Reply:
Ok got it!

Reply:
fixed